x-pack/plugin/sql/qa/server/src/main/resources/no-columns.sql-spec

```diff
@@ -0,0 +1,21 @@
+selectStar
+SELECT * FROM no_cols;
+selectStarWithFilter
+SELECT * FROM no_cols WHERE 2 > 1;
+selectStarWithFilterAndLimit
+SELECT * FROM no_cols WHERE 1 > 2 LIMIT 10;
+
+selectCount
+SELECT COUNT(*) FROM no_cols;
+// awaits fix: https://github.com/elastic/elasticsearch/issues/74311
+// selectCountWithWhere
+// SELECT COUNT(*) FROM no_cols WHERE 1 + 1 = 3;
+
+selectConst
+SELECT 1, 2, 3 FROM no_cols;
+selectConstAggregation
+SELECT MAX(1), SUM(2) FROM no_cols;
+
+// fails in H2 but cannot be tested in CSV spec because datasets without columns cannot be parsed
```

Comment:
- Just curious, why does it fail in H2?
- Maybe you can open an issue to improve testing and have a way to assert no cols in csv in the feature and link it here.

Reply:
H2 fails with a syntax error:
```
Syntax error in SQL statement "SELECT
FROM PUBLIC.""no_cols""[*]"; expected "(, USE, AS, RIGHT, LEFT, FULL, INNER, JOIN, CROSS, NATURAL, ,, SELECT"; SQL statement:
CREATE FORCE VIEW PUBLIC._0 AS
SELECT
FROM PUBLIC."no_cols" [42001-197]
```
Looks like it uses a view to execute the subselect and the creation fails because it selects an empty list. The latest version of H2 supports the query though, so https://github.com/elastic/elasticsearch/issues/39895 would resolve the issue as well.

Reply:
Thx!